src/main/java/address/updater/JarUpdater.java

```diff
@@ -128,6 +129,7 @@ private void applyUpdateToAllFiles(String sourceDir, List<String> filesToBeUpdat
      * the process has not ended yet. As such, we will make several tries with wait.
      */
     private void applyUpdate(Path source, Path dest) throws IOException {
+        //TODO: this method is about moving a file rather than applying an update?
```

Comment:
Apply update is technically replacing the old JAR with a new JAR. This method has extra check if the JAR is already accessible (Windows specific problem where JAR in process cannot be modified). Will put comments.